### DISCUSSION TITLE
v1.1.0

### DISCUSSION THREAD

**Discussion by cwilvx** [swingmx/swingmusic#112] [Announcements]
# Windows support and custom root dirs

### What's new?
This release brings the following changes to the Swing music player.

- Initial Windows support 🥳🎉🎊🎈(please provide feedback to improve the windows experience).
- You can now set a custom root directory to scan for music. Once you open the client in your browser, you'll be asked where you want to scan for music.
- Albums that contain text like: `"the essential", "best of", "greatest hits", "#1 hits", "number ones"` will be marked as compilations.
- Albums that contain text like: `"live at", "live from", "live in"` will now be marked as concert albums.
- The app automatically attempts to extract featured artists from track titles. The extracted artists are removed from the track title. You can disable this by using the `--show-feat` flag.
- Producer names are removed from track titles. eg. `What are we waiting for [prod. nick mira]`becomes `What are we waiting for` . You can disable this using the `--show-prod` flag.
- The app will automatically attempt to extract the track title and artsists from the filename if those tags are unknown. However, for now it will only work properly if the filename is in the following format: `Artist - Title`. Things like `(Official music video)` or something that starts with brackets followed by "official" will be removed.
- The app now respects the XDG Base Directory Specification. This means that the directory might be moved to `~/.config/` in linux or where your XDG variables point.
- The app now prints the location of the config folder on startup.
- The config folder directory has been renamed from `swing` to `swingmusic`.

### What's new in the Client?
- You can now use the `Esc` key to hide dialogs
- The large now playing image on the left sidebar is now animated on track change.
- Now, you'll not see an ugly blank page when there's nothing in the page. Instead, a placeholder block saying "no items in this page" will be shown.
- The page banner heights and max app width will be increased for higher resolutions. Gotta utilize that screen real estate
- Album dates are shown in artist page. All albums in artists page are also sorted by date.
- Bottom bar has been redesigned moving track information on the left, controls at the center and other actions such as repeat to the right.

- Fix: Artist tracks section not clickable properly. 
- Fix: Adding track (that already exists in queue) to the end of the queue caused it to be removed from its original position.
- Fix: Manually playing the first song playing last song on queue, when queue is set to `No-repeat`.
- Various bug fixes and improvements.

# Share and Enjoy

<hr /><em>This discussion was created from the release <a href='https://github.com/geoffrey45/swingmusic/releases/tag/v1'>v1.1.0</a>.</em>

**Comment by andy-do-dev** [comment]
Been using all morning after coming across it on the 'awsome' list. Great stuff, does all I need it to do. Have donated and will use often. Cheers.

**Reply by cwilvx** [reply]
Hello @andy-do-dev 

Thank you very much for supporting Swing Music. I really appreciate that you like my work. You just made my day.

Thank again!

**Reply by andy-do-dev** [reply]
You're welcome, @mungai-njoroge . I've played with computers since the 80's and never managed to finish anything like this. My donation is a 'hat tip' to you, sir! Well done.

I've used cmus, a cli player, for a long time but was looking for something more shiny and intuitive. This ticked the box. I intend to get to know it better, and if I can contribute in a more technical sense I will.

Good luck, and, once again, really well done.
(think I've comment on an old version... I've not commented, or put forward an issue ,on github before. Sorry if I'm doing it wrong)
